File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cursum=0
        maxi=nums[0]
        for i in nums:
            if cursum <0:
                cursum=0
            cursum = cursum + i
            maxi = max(cursum,maxi)
        return maxi

        

File: Data Structures & Algorithms/house-robber-ii/submission-0.java
class Solution {
    public int rob(int[] nums) {
        return Math.max(nums[0],Math.max(helper(Arrays.copyOfRange(nums,1,nums.length)),helper(Arrays.copyOfRange(nums,0,nums.length-1))));
    }
    public int helper(int[] nums){
        int rob2=0;
        int rob1=0;
        for(int num:nums){
            int newrob= Math.max(rob1+num,rob2);
            rob1=rob2;
            rob2=newrob;



        }
        return rob2;
    }

        
    
}
